### ISSUE TITLE
Config with ryujit but it doesnt actually use ryujit?

### ISSUE THREAD

**Issue by Barsonax** [dotnet/BenchmarkDotNet#638]
I have this config [here](https://github.com/Barsonax/Pathfindax/blob/MoreBenchmarks/Source/Code/Pathfindax.Benchmarks/BenchmarkConfigAttribute.cs)

However it doesnt actually uses ryujit when I run. Am I missing something here or is this a bug/unsupported scenario (in which case I would expect a exception). The output is very confusing as on one line it says legacyjit and another says ryujit.

This is the output I get: 
![image](https://user-images.githubusercontent.com/19387223/35611662-0eb0e9b0-0666-11e8-8be5-424e82bad5a3.png)

Changing Platform to x64 [here](https://github.com/Barsonax/Pathfindax/blob/MoreBenchmarks/Source/Code/Pathfindax.Benchmarks/BenchmarkConfigAttribute.cs#L25) does make it use ryujit. Also the perf numbers change alot which further confirms that.

**Comment by AndreyAkinshin** [comment]
@Barsonax,
Here are lines from your config:
```cs
var job = Job.Default
    .With(Platform.X86)
    .With(Jit.RyuJit);
```
.NET Framework 4.7 doesn't support RyuJIT-x86; only LegacyJIT-x86, LegacyJIT-x64, RyuJIT-x64 are supported.
I guess that BenchmarkDotNet should print a warning when the requested Job is invalid.

**Comment by Barsonax** [comment]
> I guess that BenchmarkDotNet should print a warning when the requested Job is invalid.

I would prefer that it prints a error and refuses to run in this case. Warnings might be missed.

**Comment by AndreyAkinshin** [comment]
> I would prefer that it prints a error and refuses to run in this case. Warnings might be missed.

Make sense. Then it should be a `validator`. The implementation should be similar to `CompilationValidator`.

**Comment by AndreyAkinshin** [comment]
Fixed in #693
